Source/WTF/wtf/glib/RunLoopGLib.cpp

```diff
@@ -58,7 +106,18 @@ GSourceFuncs RunLoop::s_runLoopSourceFunctions = {
         runLoopSource.runLoop->notifyEvent(RunLoop::Event::DidDispatch, name);
         return returnValue;
     },
-    nullptr, // finalize
+    // finalize
+#if HAVE(TIMERFD)
+    [](GSource* source) -> void {
+        auto& runLoopSource = *reinterpret_cast<RunLoopSource*>(source);
+        if (runLoopSource.timerFd > -1) {
+            close(runLoopSource.timerFd);
+            runLoopSource.timerFd = -1;
+        }
```

Comment:
This would be equivalent to `g_clear_fd`:

```suggestion
        g_clear_fd(&runLoopSource.timerFd, nullptr);
```

Reply:
g_clear_fd was added in GLib 2.76 whereas WebKit depends on GLib 2.70, we'd need a version bump to use it

Reply:
Let's then leave this part of the patch as-is.